### DISCUSSION TITLE
[Contract Verification] Verified contract on explorer.zksync impossible to verify on era.zksync.network (other contract)

### DISCUSSION THREAD

**Discussion by L0GYKAL** [zkSync-Community-Hub/zksync-developers#852] [Contract Verification] [answered]
### Team or Project

Zyfi

### Environment

Mainnet

### Block Explorer

https://era.zksync.network/

### Issue Type

- [ ] Plugin
- [ ] Block Explorer

### Contract Address

0x5d0d7BCa050e2E98Fd4A5e8d3bA823B49f39868d

### Compiler Type

Multi-part contract

### zkSolc Version

1.4.1

### Solc Version

0.8.22

### Contract Name

ERC1967Proxy

### Contract Code

The contract is verified on "explorer.zksync.io":
https://explorer.zksync.io/address/0x5d0d7BCa050e2E98Fd4A5e8d3bA823B49f39868d#contract#contract-info
And the code is fully available on our github repository:
https://github.com/ondefy/Zyfi-Token-Zkforge (ERC1967Proxy.sol (from Openzeppelin) and all its dependencies)
(The flag: bytecode_hash = "none" has to be used here)

### Constructor Arguments

0x000000000000000000000000ff948c6aece74108453b12e93ac8c99e75f367be0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000002426dcff15000000000000000000000000a47df9473ff4084ba4d11271ca8a470361d77a0900000000000000000000000000000000000000000000000000000000

From https://era.zksync.network/tx/0x80571d7a0f39717cd56eb4df9962454e7694f9b2c0648d19e3e1416e4f592186

### Repo Link (Optional)

https://github.com/ondefy/Zyfi-Token-Zkforge

### Additional Details

The contract is already deployed and verified on "explorer.zksync.io" at this address:
https://explorer.zksync.io/address/0x5d0d7BCa050e2E98Fd4A5e8d3bA823B49f39868d#transactionsThe flag: bytecode_hash = "none" has to be used here.

**Comment by alhenaai** [comment]
To verify your `ERC1967Proxy` contract on the ZKsync Era network at `https://era.zksync.network/`, you should ensure that the bytecode and constructor arguments match exactly with those used during the deployment. Since the contract is already verified on `https://explorer.zksync.io/`, the issue might be related to the specific settings or environment differences between the two explorers.

Here are a few steps you can follow to troubleshoot and potentially resolve the verification issue on the ZKsync Era network:

1. **Ensure Correct Bytecode**: Double-check that the bytecode generated for verification matches the deployed bytecode. This includes ensuring that all compiler settings (like optimization and the `bytecode_hash = "none"` flag) are identical.

2. **Constructor Arguments**: Verify that the constructor arguments used during the deployment are correctly formatted and encoded for the ZKsync Era verification process.

3. **Compiler Version Consistency**: Make sure that the `zkSolc` version (1.4.1) and `Solc` version (0.8.22) used for compilation are supported and consistent across both verification platforms.

4. **Check Contract Dependencies**: Since `ERC1967Proxy` is a part of a multi-part contract, ensure that all dependent contracts are also correctly verified or accounted for in the verification process.

5. **Use the Correct Verification Plugin**: If you are using a Hardhat plugin for verification, ensure it is compatible with the ZKsync Era network. Update the plugin if necessary.

If after following these steps you still face issues, it might be beneficial to consult the detailed documentation or community discussions for specific guidance related to contract verification on the ZKsync Era network. You can find more in-depth discussions and support on the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by uF4No** [comment, marked as answer]
Proxy contract was verified by etherscan: https://era.zksync.network/address/0x5d0d7BCa050e2E98Fd4A5e8d3bA823B49f39868d#code
